pallets/loan/src/types.rs

```diff
@@ -110,12 +112,35 @@ where
 	/// returns the present value of the loan
 	/// note: this will use the accumulated_rate and last_updated from self
 	/// if you want the latest upto date present value, ensure these values are updated as well before calling this
-	pub(crate) fn present_value(&self) -> Option<Amount> {
+	pub(crate) fn present_value(
+		&self,
+		write_off_groups: &Vec<WriteOffGroup<Rate>>,
+	) -> Option<Amount> {
 		// calculate current debt and present value
-		math::debt(self.principal_debt, self.accumulated_rate).and_then(|debt| {
-			self.loan_type
-				.present_value(debt, self.last_updated, self.rate_per_sec)
-		})
+		math::debt(self.principal_debt, self.accumulated_rate)
+			.and_then(|debt| {
+				// if the debt is written off, write off accordingly
+				match self.write_off_index {
```

Comment:
This should probably just be a `self.write_off_index.map_or(debt, |index| ....)`

Reply:
I like it